src/McBopomofo.cpp

```diff
@@ -215,6 +243,20 @@ void McBopomofoEngine::activate(const fcitx::InputMethodEntry&,
 
 void McBopomofoEngine::reset(const fcitx::InputMethodEntry&,
                              fcitx::InputContextEvent& event) {
+  auto isFocusOutEvent = dynamic_cast<fcitx::FocusOutEvent*>(&event) != nullptr;
+
+  // Note: when inputting in Chrome based web browsers, the `reset` method would
+  // be called when the input method commits evicted text by passing a focus out
+  // event. We have to ignore such events otherwise we are hardly to input a
+  // complete Chinese sentense.
+  if (isFocusOutEvent) {
+    auto context = event.inputContext();
+    auto program = context->program();
+    if (program.rfind("google-chrome", 0) == 0) return;
```

Comment:
Can we make this a user configurable list?

Reply:
Sounds like a good idea. Let me do this later.